ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFFromUnixTime.java

```diff
@@ -153,23 +127,14 @@ public Object evaluate(DeferredObject[] arguments) throws HiveException {
       unixtime = inputLongOI.get(arguments[0].get());
     }
 
-    Date date = new Date(unixtime * 1000L);
-    result.set(formatter.format(date));
+    Instant i = Instant.ofEpochSecond(unixtime);
```

Comment:
int is used for seconds and long is used for milliseconds. Put a check for the same and call Instant.ofEpochSecond() and Instant.ofEpochMili() respectively